Sources/NostrSDK/Events/SetMetadataEvent.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  SetMetadataEvent.swift
+//  
+//
+//  Created by Bryan Montz on 7/22/23.
+//
+
+import Foundation
+
+/// An object that describes a user.
+public struct UserMetadata: Decodable {
+    
+    /// The user's name.
+    public let name: String?
+    
+    /// The user's description of themself.
+    public let about: String?
+    
+    /// The user's website address.
+    public let website: URL?
```

Comment:
Oh, interesting. The decoder is smart enough to transform strings into URLs and vice-versa?

Reply:
This surprised me too, but yeah!